modules/influxdb/influxdb.go

```diff
@@ -57,39 +73,23 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 			// "Open shard" which is the last thing that happens before the server is ready to accept connections.
 			// This is probably different for InfluxDB 2.x, but that is left as an exercise for the reader.
 			strategies := []wait.Strategy{
-				genericContainerReq.WaitingFor,
+				wait.ForListeningPort("8086/tcp"),
 				wait.ForLog("influxdb init process in progress..."),
 				wait.ForLog("Server shutdown completed"),
 				wait.ForLog("Opened shard"),
 			}
-			genericContainerReq.WaitingFor = wait.ForAll(strategies...)
-			hasInitDb = true
-			break
+			return wait.ForAll(strategies...)
 		}
 	}
 
-	if !hasInitDb {
-		if lastIndex := strings.LastIndex(genericContainerReq.Image, ":"); lastIndex != -1 {
-			tag := genericContainerReq.Image[lastIndex+1:]
-			if tag == "latest" || tag[0] == '2' {
-				genericContainerReq.WaitingFor = wait.ForLog(`Listening log_id=[0-9a-zA-Z_~]+ service=tcp-listener transport=http`).AsRegexp()
+	return wait.ForHTTP("/health").
+		WithResponseMatcher(func(body io.Reader) bool {
+			bs, err := io.ReadAll(body)
+			if err != nil {
+				return false
 			}
-		} else {
-			genericContainerReq.WaitingFor = wait.ForLog("Listening for signals")
-		}
-	}
-
-	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
-	var c *InfluxDbContainer
-	if container != nil {
-		c = &InfluxDbContainer{Container: container}
-	}
-
-	if err != nil {
-		return c, fmt.Errorf("generic container: %w", err)
-	}
-
-	return c, nil
+			return strings.Contains(string(bs), "ready for queries and writes")
```

Comment:
I think the check for `"status": "pass",` could be more robust, but I do not have a strong opinion on which one is better.

Reply:
I was undecided on which approach to use, so for simplicity, I used the same one as the health endpoint in `modules/openfga/openfga.go:61`. However, I agree that checking the JSON status for a pass value might be more appropriate. I’ll update it accordingly.